src/components/offerStatus/StatusFormOnPage.js

```diff
@@ -0,0 +1,120 @@
+
+const StatusFormOnPage = ({ offer, error }) => {
+  const { t, i18n } = useTranslation();
+  const router = useRouter();
+
+  const offerId = parseOfferId(offer['@id']);
+  const offerType = parseOfferType(offer['@context']);
+  const name =
+    offer?.name?.[i18n.language] ?? offer?.name?.[offer.mainLanguage];
+  const rawStatusType = offer?.status?.type;
+  const rawStatusReason = offer?.status?.reason;
+
+  const [type, setType] = useState('');
+  const [reason, setReason] = useState('');
+
+  useEffect(() => {
+    if (!rawStatusType) return;
+    setType(rawStatusType);
+  }, [rawStatusType]);
+
+  useEffect(() => {
+    const newReason = offer?.status?.reason?.[i18n.language];
+    if (!rawStatusReason || !newReason) return;
+    setReason(newReason);
+  }, [rawStatusReason]);
+
+  const useChangeStatus = useMemo(() => {
+    if (offerType === OfferType.PLACE) {
+      return useChangeStatusPlace;
+    }
+    return useChangeStatusEvent;
+  }, [offerType]);
+
+  const handleSuccessChangeStatus = () =>
+    router.push(`/${offerType}/${offerId}/preview`);
+
+  const { mutate: changeStatus, ...changeStatusMutation } = useChangeStatus({
+    onSuccess: handleSuccessChangeStatus,
+  });
+
+  return (
+    <Page>
+      <Page.Title>{t('offerStatus.title', { name })}</Page.Title>
+      <Page.Content spacing={5}>
+        {changeStatusMutation.status === QueryStatus.LOADING ? (
+          <Spinner marginTop={4} />
+        ) : error || changeStatusMutation.error ? (
+          <Alert variant={AlertVariants.WARNING}>
+            {error.message || changeStatusMutation.error?.message}
+          </Alert>
+        ) : (
+          [
+            <ReasonAndTypeForm
+              key="reason-and-type"
+              offerType={offerType}
+              statusType={type}
+              statusReason={reason}
+              onChangeStatusType={(e) => setType(e.target.value)}
+              onInputStatusReason={(e) => setReason(e.target.value)}
+            />,
+            <Inline key="actions" spacing={3}>
+              <Button
+                variant={ButtonVariants.SECONDARY}
+                onClick={() => router.push(`/offer/${offerId}/edit`)}
+              >
+                {t('offerStatus.actions.cancel')}
+              </Button>
+              <Button
+                variant={ButtonVariants.PRIMARY}
+                disabled={
+                  !offer ||
+                  reason.length === 0 ||
+                  reason.length > MaxLengthReason
+                }
+                onClick={() => {
+                  changeStatus({
+                    id: offerId,
+                    type,
+                    reason:
+                      type === OfferStatus.AVAILABLE
+                        ? undefined
+                        : {
+                            ...offer.status.reason,
+                            [i18n.language]: reason,
+                          },
+                  });
+                }}
+              >
```

Comment:
Maybe best to check with @corneelwille , but I have a feeling the save button should go first in this case.

<img width="1609" alt="Screen Shot 2021-02-22 at 17 06 58" src="https://user-images.githubusercontent.com/959026/108735112-71058800-7530-11eb-9a37-11a8388aeeeb.png">

In modals the primary button is on the right side, but the buttons are also aligned to the right there. Here they are aligned to the left. I don't think we have an explicit rule about this, but looking at other apps it seems like the primary button is always the first when aligned left and last when aligned right.

Reply:
Also, can we set a max-width on the form? The optimal line length of text is usually between 50-75 characters per line. (For example https://baymard.com/blog/line-length-readability)

As you can see in the screenshot, the text gets very wide on bigger screens and harder to read.

Reply:
done